server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailSpooler.java

```diff
@@ -61,35 +62,18 @@
      */
     private int numThreads;
 
-    /**
-     * Number of active threads
-     */
-    private final AtomicInteger numActive = new AtomicInteger(0);
-
     private final AtomicInteger processingActive = new AtomicInteger(0);
 
-    /**
-     * Spool threads are active
-     */
-    private final AtomicBoolean active = new AtomicBoolean(false);
-
     private final MetricFactory metricFactory;
 
-    /**
-     * Spool threads
-     */
-    private ExecutorService dequeueService;
-
-    private ExecutorService workerService;
-
     /**
      * The mail processor
      */
     private MailProcessor mailProcessor;
 
     private MailQueueFactory<?> queueFactory;
-
-    private int numDequeueThreads;
+    private reactor.core.Disposable disposable;
```

Comment:
Can't you add it in `imports` (already used?)

Reply:
Yes already used by James disposable API

Reply:
there's already 
```java
import org.apache.james.lifecycle.api.Disposable;
```